### ISSUE TITLE
False negative for D417

### ISSUE THREAD

**Issue by PhilReinhold** [charliermarsh/ruff#1406]
It appears that ruff v0.0.195 fails to detect missing argument docs from a google-formatted docstring (D417) when a newline is present in the `Returns` block.

```python
# test.py
def f(x, y, z):
    """Do f.

    Args:
        x: the value
        z: A final argument

    Returns: 
        the value
    """
    return x

def g(x, y, z):
    """Do g.

    Args:
        x: the value
        z: A final argument

    Returns: the value
    """
    return x

def h(x, y, z):
    """Do h.

    Args:
        x: the value
        z: A final argument
    """
    return x
```

```
$ ruff --version
ruff 0.0.195

$ ruff --select D --show-source test.py
test.py:14:1: D417 Missing argument description in the docstring: `y`
   |
14 | / def g(x, y, z):
15 | |     """Do g.
16 | |
17 | |     Args:
18 | |         x: the value
19 | |         z: A final argument
20 | |
21 | |     Returns: the value
22 | |     """
23 | |     return x
   | |____________^ D417
   |
test.py:25:1: D417 Missing argument description in the docstring: `y`
   |
25 | / def h(x, y, z):
26 | |     """Do h.
27 | |
28 | |     Args:
29 | |         x: the value
30 | |         z: A final argument
31 | |     """
32 | |     return x
   | |____________^ D417
   |
Found 2 error(s).
```
I'm expecting 3 errors for all three functions, not just g and h.

(I promise I don't just sit around looking for corner-case D417 bugs! But checking for missing docstring args is one of the useful features of pydocstyle which I'm hoping ruff can replace)

**Comment by charliermarsh** [comment]
I’ve gotta see if the latest pydocstyle catches this. Wondering if this was a regression from when I last updated the parser to match their logic.

**Comment by charliermarsh** [comment]
Yeah pydocstyle doesn't catch this either.

**Comment by charliermarsh** [comment]
I'm going to make the improvement that pydocstyle described [here](https://github.com/PyCQA/pydocstyle/issues/459), to use a dedicated convention flag. pydocstyle seems to have stalled, I don't see much movement there, but I can implement this for sure in Ruff.

**Comment by PhilReinhold** [comment]
Yes, official "docstring-convention" support would be a wonderful feature! I guess I can feel safe replacing pydocstyle if it has the same bugs.

**Comment by charliermarsh** [comment]
This'll go out today, working on it now, just a sec.
